fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,229 @@
+            assertEquals(500, expectedKeys.size());
+            assertIndexEntryPrimaryKeyTuples(expectedKeys,
+                    store.scanIndex(dataModel.index, scanBounds, null, ScanProperties.FORWARD_SCAN));
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void scanWithContinuationsTest(boolean isGrouped) throws Exception {
+        final long seed = 85373450L;
+        final boolean isSynthetic = false;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(210)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(500, start, context, 2);
+            commit(context);
+        }
+        // Scan properties with a limit of 36
+        final ScanProperties scanProperties = new ScanProperties(ExecuteProperties.newBuilder()
+                .setReturnedRowLimit(36)
+                .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                .build());
+
+        // This test only uses the match-all query
+        LuceneQueryClause search = LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY;
+
+        try (FDBDatabaseRunner runner = fdb.newRunner()) {
+            // Create a cursor that reads through all continuations
+            RecordCursor<IndexEntry> cursor = new AutoContinuingCursor<>(
+                    runner,
+                    (context, continuation) -> {
+                        final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+                        LuceneScanBounds scanBounds = isGrouped
+                                                      ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                                      : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+                        return store.scanIndex(dataModel.index, scanBounds, continuation, scanProperties);
+                    });
+            // Run the scan with the given query and assert the results
+            final Tuple groupTuple = LuceneIndexTestDataModel.calculateGroupTuple(isGrouped, 2);
+            final Set<Tuple> expectedKeys = dataModel.groupingKeyToPrimaryKeyToPartitionKey.get(groupTuple).keySet();
+            assertEquals(500, expectedKeys.size());
+            assertIndexEntryPrimaryKeyTuples(expectedKeys, cursor);
+        }
+    }
+
+
+    @Nonnull
+    private static Tuple saveRecord(final boolean isGrouped, final boolean isSynthetic, final LuceneIndexTestDataModel dataModel, final long start, final FDBRecordStore store, final int group) {
```

Comment:
Just poking at `LuceneIndexTestDataModel` independently, making these public, instance methods on `LuceneIndexDataModel` would make a lot of sense. `isGrouped` and `isSynthetic` can be fetched from there, so you just end up with something like:
```
dataModel.saveRecord(store, start, group);
dataModel.saveEmptyRecord(store, start, group);
```

Reply:
Sure, done.